src/borrower/feed/navfeed.sol

```diff
@@ -189,22 +190,46 @@ abstract contract NAVFeed is Auth, Discounting {
 
         bytes32 nftID_ = nftID(loan);
         WriteOffGroup memory writeOffGroup_ = writeOffGroups[writeOffGroupIndex_];
-        require(block.timestamp >= maturityDate[nftID_] + writeOffGroup_.overdueDays, "too-early");
+        uint maturityDate_ = maturityDate[nftID_];
+        require(block.timestamp >= maturityDate_ + writeOffGroup_.overdueDays * 60 * 60 * 24, "too-early");
 
         uint currentRate = pile.loanRates(loan);
-        require(writeOffGroup_.percentage.value > writeOffGroups[currentRate].percentage.value, "cannot-decrease-writeoff");
+        require(currentRate < WRITEOFF_RATE_GROUP_START || writeOffGroup_.overdueDays > writeOffGroups[currentRate].overdueDays, "cannot-go-back");
+
+        if (pile.loanRates(loan) < WRITEOFF_RATE_GROUP_START) {
+            // this is the first writeoff, remove future value for loan from bucket
+            uint fv = futureValue[nftID_];
+            buckets[maturityDate_] = safeSub(buckets[maturityDate_], fv);
+
+            if (uniqueDayTimestamp(lastNAVUpdate) >= maturityDate_) {
+                overdueButNotWrittenOff = secureSub(overdueButNotWrittenOff, fv);
+            }
+        }
 
         pile.changeRate(loan, WRITEOFF_RATE_GROUP_START + writeOffGroupIndex_);
     }
 
     function overrideWriteOff(uint loan, uint writeOffGroupIndex_) public auth {
         writeOffOverride[loan] = true;
+
+        if (pile.loanRates(loan) < WRITEOFF_RATE_GROUP_START) {
+            // this is the first writeoff, remove future value for loan from bucket
+            bytes32 nftID_ = nftID(loan);
+            uint maturityDate_ = maturityDate[nftID_];
+            uint fv = futureValue[nftID_];
+            buckets[maturityDate_] = safeSub(buckets[maturityDate_], fv);
+
+            if (uniqueDayTimestamp(lastNAVUpdate) >= maturityDate_) {
+                overdueButNotWrittenOff = safeSub(overdueButNotWrittenOff, fv);
+            }
+        }
+
         pile.changeRate(loan, WRITEOFF_RATE_GROUP_START + writeOffGroupIndex_);
     }
 
     // --- NAV calculation ---
     function currentNAV() public view returns(uint) {
-        return safeAdd(currentDiscount(), currentWriteOffs());
+        return safeAdd(safeAdd(currentDiscount(), overdueButNotWrittenOff), currentWriteOffs());
```

Comment:
This would break our current pattern, if I understand correctly.
Currently the view method grantees a up to date NAV.

I think we should calculate the new `overdueButNotWrittenOff` value.